### PR TITLE
Reduce server config to only bootstrapping of user and credentials

### PR DESCRIPTION
**Pull request by dnephin**
## Summary

Implements most of what was proposed in #3915. 

Not done, still pending further discussion:
* `User.Password` has not been removed - it's used by helm
* `SupportAdminGroups` has not been added yet
* the access key expiration time is still set to 10 years

I'd like to address those in separate follow up PRs. 

If a self-hosted install of Infra includes `grants` or `providers`  the server will refuse to start and the error message will inform the user what needs to change.

Helm changes in https://github.com/infrahq/helm-charts/pull/17

### CODE REVIEW

File: internal/server/data/grant.go
Comment (BruceMacD) [inline internal/server/data/grant.go:276]: nice, glad to see this get cleaned up

**Review by BruceMacD — APPROVED** [review]

File: internal/server/config.go
Comment (dnephin) [reply internal/server/config.go:33]: Ya, I like that! I'll change it to `InfraRole`

File: internal/cmd/server_test.go
Comment (jmorganca) [inline internal/cmd/server_test.go:132]: Fantastic clean up in this PR. It's most likely better to visit this in a follow up PR, but could we further simplify this by removing `users:`? (can move this to a new issue if so)

On first thought, a minimal bootstrap interface could be:

1. A single `accessKey` field for bootstrapping the system via terraform and for follow-on support admin needs
2. A username+password (for bootstrapping via the dashboard) although I don't know if this is a great experience. Quite a few tools would allow just allow the user to sign up as usual (of course would require multi-org by default) after standing up the server by visiting the endpoint it exposes in the browser

**Comment by dnephin** [comment]
Helm chart changes are in https://github.com/infrahq/helm-charts/pull/17. I need to test those first before merging

**Comment by jmorganca** [comment]
LGTM once any dependent changes are in! (E.g. helm)

**Comment by dnephin** [comment]
From https://github.com/infrahq/helm-charts/pull/17#issuecomment-1376196751, it sounds like we probably need to merge this first. The helm changes would come after we have a version of the server image to tag.

**Comment by jmorganca** [comment]
That makes sense, I had it backwards. LGTM!

**Review by jmorganca — APPROVED** [review]

**Pull request merged**
